src/main/java/com/sproutt/eussyaeussyaapi/api/mission/MissionController.java

```diff
@@ -0,0 +1,164 @@
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.time.Instant;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class MissionController {
+
+    @Value("${jwt.header}")
+    private String tokenKey;
+
+    @Value("${jwt.secret}")
+    private String secretKey;
+
+    private static final String ZONE_SEOUL = "Asia/Seoul";
+    private static final LocalTime START_AVAILABLE_LOCAL_TIME = LocalTime.of(4, 0);
+    private static final LocalTime END_AVAILABLE_LOCAL_TIME = LocalTime.of(9, 0);
+
+    private final MemberService memberService;
+    private final MissionService missionService;
+    private final JwtHelper jwtHelper;
+
+    @PostMapping("/missions")
+    public ResponseEntity createMission(@RequestHeader HttpHeaders requestHeaders, @RequestBody @Valid MissionDTO missionDTO) {
+        String token = requestHeaders.get(tokenKey).get(0);
+        Member loginMember = getTokenOwner(token);
+
+        if (!isAvailableTime(requestHeaders.getDate())) {
+            throw new NotAvailableTimeException();
+        }
+
+        missionService.create(loginMember, missionDTO);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        return new ResponseEntity<>(headers, HttpStatus.CREATED);
+    }
+
+    @GetMapping("/missions/{missionId}")
+    public ResponseEntity<Mission> searchMission(@PathVariable Long missionId) {
+        Mission mission = missionService.findById(missionId);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        return new ResponseEntity<>(mission, headers, HttpStatus.OK);
+    }
+
+    @GetMapping("/missions")
+    public ResponseEntity<List<Mission>> searchMissionList(@RequestParam(name = "writer", required = false) String memberId, @RequestParam(name = "after", required = false) String afterDate, @RequestParam(name = "before", required = false) String beforeDate) {
+        List<Mission> missionList;
+
+        if (memberId != null) {
+            Member writer = memberService.findByMemberId(memberId);
+            missionList = missionService.findByWriter(writer);
+        } else {
+            missionList = missionService.findAll();
+        }
+
+        missionList = missionService.filterDate(afterDate, beforeDate, missionList);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        return new ResponseEntity<>(missionList, headers, HttpStatus.OK);
```

Comment:
```suggestion
        HttpHeaders headers = new HttpHeaders();
        headers.setContentType(MediaType.APPLICATION_JSON);

		if(memberId == null) {
			return new ResponseEntity<>(missionService.findAll(), headers, HttpStatus.OK);
		}
		
		var writer = memberService.findByMemberId(memberId);
        var missions = missionService.findByWriter(writer);
        
        var rangedMissions = missionService.filterDate(afterDate, beforeDate, missions);
       
        return new ResponseEntity<>(rangedMissions, headers, HttpStatus.OK);
```

early return 을 활용한 이런 코드는 어떠신가요